src/libpriv/rpmostree-core.c

```diff
@@ -2860,21 +2861,51 @@ handle_file_dispositions (RpmOstreeContext *self,
   return TRUE;
 }
 
+typedef struct {
+  GHashTable *files_skip;
+  GStrv files_remove_regex;
+} FilterData;
+
 static OstreeRepoCheckoutFilterResult
 checkout_filter (OstreeRepo         *self,
                  const char         *path,
                  struct stat        *st_buf,
                  gpointer            user_data)
 {
-  GHashTable *files_skip = user_data;
-  if (g_hash_table_contains (files_skip, path))
-    return OSTREE_REPO_CHECKOUT_FILTER_SKIP;
+  GHashTable *files_skip = ((FilterData*)user_data)->files_skip;
+  GStrv files_remove_regex = ((FilterData*)user_data)->files_remove_regex;
+
+  if (files_skip && g_hash_table_size (files_skip) > 0)
+    {
+      if (g_hash_table_contains (files_skip, path))
+        return OSTREE_REPO_CHECKOUT_FILTER_SKIP;
+    }
+
+  if (files_remove_regex)
+    {
+      guint len = g_strv_length(files_remove_regex);
+      for (guint i = 0; i < len; i++)
+        {
+          const char *remove_regex_pattern = files_remove_regex[i];
+          GRegex *regex = g_regex_new (remove_regex_pattern, G_REGEX_JAVASCRIPT_COMPAT, 0, NULL);
+          if (!regex)
+            continue;
```

Comment:
Hmm, sorry I should've spotted this earlier but: this is recompiling the regex each time. So that means that if a package has 100 files, we're recompiling every pattern 100 times, which is not efficient. Also it looks like we're leaking the `GRegex` object here (and so was the previous code).

WDYT about making `files_remove_regex` instead be a `GPtrArray` of `GRegex` objects?

Reply:
Ohh true. Yes, that would be much more efficient.